packages/ui/src/components/va-counter/VaCounter.vue

```diff
@@ -0,0 +1,368 @@
+        <va-icon
+          class="va-counter__icon-increase"
+          v-bind="increaseIconProps"
+        />
+      </slot>
+    </template>
+
+    <template v-if="$slots.content" #content="slotScope">
+      <div ref="input" tabindex="0">
+        <slot name="content" v-bind="{ ...slotScope, value: valueComputed }" />
+      </div>
+    </template>
+
+    <input
+      v-if="!$slots.content"
+      class="va-input__content__input"
+      ref="input"
+      type="number"
+      v-bind="{ ...computedInputAttributes, ...inputListeners }"
+      :value="valueComputed"
+      @input="onManualInput($event.target.value)"
+      @change="onManualChange($event.target.value)"
+    >
+  </VaInputWrapper>
+</template>
+
+<script lang="ts">
+import { computed, defineComponent, InputHTMLAttributes, ref, PropType } from 'vue'
+import { useFormProps } from '../../composables/useForm'
+import { useEmitProxy } from '../../composables/useEmitProxy'
+import { useFocus, useFocusEmits } from '../../composables/useFocus'
+import { useStatefulProps, useStateful } from '../../composables/useStateful'
+import { useColor } from '../../composables/useColor'
+import VaInputWrapper from '../va-input/components/VaInputWrapper.vue'
+import VaIcon from '../va-icon/VaIcon.vue'
+import VaButton from '../va-button/VaButton.vue'
+import { omit, pick } from 'lodash-es'
+
+const { createEmits: createInputEmits, createListeners: createInputListeners } = useEmitProxy(
+  ['change'],
+)
+
+const { createEmits: createFieldEmits, createListeners: createFieldListeners } = useEmitProxy([
+  { listen: 'click-prepend', emit: 'click:decrease-button' },
+  { listen: 'click-append', emit: 'click:increase-button' },
+  { listen: 'click-prepend-inner', emit: 'click:decrease-icon' },
+  { listen: 'click-append-inner', emit: 'click:increase-icon' },
+])
+
+export default defineComponent({
+  name: 'VaCounter',
+
+  components: { VaInputWrapper, VaIcon, VaButton },
+
+  props: {
+    ...useFormProps,
+    ...useStatefulProps,
+    // input
+    modelValue: { type: [String, Number], default: '' },
+    manualInput: { type: Boolean, default: false },
+    stateful: { type: Boolean, default: false },
+    min: { type: Number, default: 1 },
+    max: { type: Number, default: undefined },
+    step: { type: Number, default: 1 },
+    label: { type: String, default: '' },
+    // hint
+    messages: { type: [Array, String] as PropType<string[] | string>, default: () => [] },
+    // style
+    width: { type: String, default: '154px' },
+    color: { type: String, default: 'primary' },
+    outline: { type: Boolean, default: undefined },
+    bordered: { type: Boolean, default: undefined },
+    // icons & buttons
+    increaseIcon: { type: String, default: 'add' },
+    decreaseIcon: { type: String, default: 'remove' },
+    buttons: { type: Boolean, default: false },
+    margins: { type: [String, Number], default: '4px' },
+    flat: { type: Boolean as PropType<boolean>, default: true },
```

Comment:
Not a fun of default true values. Could be named "filled" which defaults to false so you can simply pass "filled" flag instead of 'flat="false"' expression